internal/srv/server.go

```diff
@@ -50,17 +53,17 @@ func (s *Server) Run(ctx context.Context) error {
 
 	s.Logger.Infow("starting subscribers")
 
-	if err := s.configureSubscribers(); err != nil {
+	if err := s.configureSubscribers(ctx); err != nil {
 		s.Logger.Errorw("unable to configure subscribers", "error", err)
 		return err
 	}
 
 	for _, ch := range s.changeChannels {
-		go s.processChange(ch)
+		go s.listenChange(ctx, ch)
 	}
 
 	for _, ev := range s.eventChannels {
-		go s.processEvent(ev)
+		go s.listenEvent(ctx, ev)
 	}
```

Comment:
it looks like you don't need to pass the viper `ctx` through to these functions, as it is already set and available in the `server`.